sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -622,7 +622,8 @@ Exceptions:
 
 Memory that is allocated by the following functions is owned by the specific
 graph. When freed inside the graph, the memory is only accessible before the
-`free` node is executed and after the `malloc` node is executed.
+`free` node is executed and after the `malloc` node is executed. When a `command_graph` contains
+any nodes that allocate memory, only one executable graph object can be alive at the same time.
```

Comment:
A note that CUDA Graphs also has the limitation that if a graph contains a memory allocation node, then only one instantiation of the graph can exist at any point in time.

Reply:
Not only the outside access of allocations is the issue.
Since the pointer is defined outside the graph it would be a shared resource among executable graph objects and its command functions. Unless we change the semantics away from plain pointers such a concurrent access would be
leading to undefined behavior when we allow more than one instantiation.

Reply:
> Not only the outside access of allocations is the issue. Since the pointer is defined outside the graph it would be a shared resource among executable graph objects and its command functions. Unless we change the semantics away from plain pointers such a concurrent access would be leading to undefined behavior when we allow more than one instantiation.

I don't think is necessarily true. At the end of the day the command function is being compiled to an actual device kernel which takes parameters and we can set whatever value for those parameters that we want before submission, typically they will be what has been pulled from the CGF itself but that doesn't have to be the case. Obviously this is not true of the current prototype because we're just reusing the CGF but as documented in the issues this has many limitations already. 

One caveat though actually is that this probably isn't true for `host_task`s when we eventually support them, since it's likely just re-running the actual host lambda/function so would use captured values directly. In theory I suppose the compiler could output information about host task functions in the integration header and then we could tinker with them, though I'm not a compiler person, and even then modifying a lambda in this way is probably not allowed or at the very least highly discouraged.

Reply:
Another point Re: the discussion about device only allocations inside a graph, although it's safe to say that user couldn't do anything with a device only pointer on the host inside a `host_task` they could actually use it for interop with the backend and so at that point each graph having it's own device allocations would be very problematic as described here.

I think on balance then if we want to support allocation nodes then the single executable graph restriction makes a lot of sense. Feel free to mark this comment as resolved if there's no further discussion to be had!